### PR TITLE
Add go-counting exercise

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: exercises/go-counting/Example.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            new Point(col, row - 1),
+            new Point(col-1, row),
+            new Point(col+1, row),
+            new Point(col, row+1)
+        };
+
+        return coords.Where(IsValidCoordinate);
+    }
+
+    private IEnumerable<Point> NonEmptyNeighborCoordinates(Point coordinate) =>
```
Comment (robkeim) [inline exercises/go-counting/Example.cs]: To be consistent with the terminology above this should be `TakenNeighborCoordinates` to match with `isTaken`/`isEmpty`

Comment (robkeim) [inline exercises/go-counting/Example.cs]: @ErikSchierboom I've left one easy to address comment, and then I'll merge.

Comment (ErikSchierboom) [inline exercises/go-counting/Example.cs]: Great point! I've updated the code and also renamed `isTaken` and `isEmpty` to `IsTaken` and `IsEmpty` and then used `IsTaken` in `TakenNeighborCoordinates` and removed the unnecessary call to `IsTaken` in `TerritoryOwner`.

**Comment by robkeim** [comment]
Great work, thanks @ErikSchierboom!

**Pull request merged**